src/main/kotlin/app/extractors/CommonExtractor.kt

```diff
@@ -10,52 +10,63 @@ class CommonExtractor : ExtractorInterface {
+            reversedMap["actionscript"] = listOf("as")
+            reversedMap["arduino"] = listOf("ino")
+            reversedMap["assembly"] = listOf("asm", "s", "S")
+            reversedMap["clojure"] = listOf("clj", "cljs", "cljc", "edn")
+            reversedMap["cobol"] = listOf("cbl", "cob", "cpy")
+            reversedMap["coffeescript"] = listOf("coffee", "litcoffee")
+            reversedMap["cuda"] = listOf("cu", "cuh")
+            reversedMap["d"] = listOf("d")
+            reversedMap["dosbatch"] = listOf("bat")
+            reversedMap["emacslisp"] = listOf("el", "elc")
+            reversedMap["erlang"] = listOf("erl", "hrl")
+            reversedMap["elm"] = listOf("elm")
+            reversedMap["factor"] = listOf("factor")
+            reversedMap["forth"] = listOf("forth", "4TH")
+            reversedMap["fortran"] = listOf("f", "for", "f90", "f95", "f03",
+                    "f08", "f15")
+            reversedMap["gradle"] = listOf("gradle")
+            reversedMap["groovy"] = listOf("groovy")
+            reversedMap["haskell"] = listOf("hs", "lhs")
+            reversedMap["haxe"] = listOf("hx")
+            reversedMap["html"] = listOf("html", "htm")
+            reversedMap["hy"] = listOf("hy")
+            reversedMap["j"] = listOf("ijs")
+            reversedMap["julia"] = listOf("jl")
+            reversedMap["lisp"] = listOf("lisp", "lsp", "l")
+            reversedMap["lua"] = listOf("lua")
+            reversedMap["makefile"] = listOf("makefile")
+            reversedMap["matlab"] = listOf("m", "mlx")
+            reversedMap["maven"] = listOf("pom")
+            reversedMap["ocaml"] = listOf("ml", "mli")
+            reversedMap["oxygene"] = listOf("oxygene")
+            reversedMap["pascal"] = listOf("pas")
+            reversedMap["perl"] = listOf("pl", "PL")
+            reversedMap["powershell"] = listOf("ps1", "psm1", "psd1")
+            reversedMap["processing"] = listOf("pde")
+            reversedMap["prolog"] = listOf("pl", "P")
+            reversedMap["puppet"] = listOf("pp")
+            reversedMap["r"] = listOf("r", "R")
+            reversedMap["rust"] = listOf("rs")
+            reversedMap["sas"] = listOf("sas")
+            reversedMap["scala"] = listOf("scala", "sc")
+            reversedMap["scheme"] = listOf("scm", "ss")
+            reversedMap["shell"] = listOf("sh")
+            reversedMap["smalltalk"] = listOf("st")
+            reversedMap["sql"] = listOf("sql")
+            reversedMap["tcl"] = listOf("tcl")
+            reversedMap["tex"] = listOf("tex")
+            reversedMap["typescript"] = listOf("ts", "tsx")
+            reversedMap["verilog"] = listOf("v")
+            reversedMap["vhdl"] = listOf("vhdl")
+            reversedMap["viml"] = listOf("vim")
+            reversedMap["visualbasic"] = listOf("bas")
+            reversedMap["visualbasicforapps"] = listOf("vba")
+            reversedMap["visualprolog"] = listOf("pro")
```

Comment:
Just noticed. Please, remove "pro" extension, as it conflicts with C project files and users assigned wrong Prolog stats.